src/targets/maven.ts

```diff
@@ -444,4 +391,105 @@ export class MavenTarget extends BaseTarget {
 
     return `${moduleName}.jar`;
   }
+
+  // Maven central does not indicate when it completes the action, so we need to
+  // retry every so often and query it for the new state of repository.
+  // Based on: https://github.com/vanniktech/gradle-maven-publish-plugin/ implementation.
+  public async closeAndReleaseRepository(): Promise<void> {
+    const repository = await this.getRepository();
+    const { repositoryId, type } = repository;
+
+    if (type !== 'open') {
+      throw new Error(
+        'No open repositories available. Go to Nexus Repository Manager to see what happened.'
+      );
+    }
+
+    await this.closeRepository(repositoryId);
+    await this.releaseRepository(repositoryId);
+  }
+
+  private getNexusRequestHeaders(): Record<string, string> {
+    return {
+      Accept: 'application/json',
+      Authorization: `Basic ${Buffer.from(
+        `${process.env.OSSRH_USERNAME}:${process.env.OSSRH_USERNAME}`
+      ).toString(`base64`)}`,
+    };
+  }
+
+  private async getRepository(): Promise<NexusRepository> {
+    const response = await fetch(`${NEXUS_API_BASE_URL}/profile_repositories`, {
+      headers: this.getNexusRequestHeaders(),
+    });
+    if (!response.ok) {
+      throw new Error(
+        `Unable to fetch repository: ${response.status}, ${response.statusText}`
+      );
+    }
+
+    const body = await response.json();
+    const repositories = body.data;
+
+    if (repositories.length === 0) {
+      throw new Error(`No available repositories. Nothing to publish.`);
+    }
+
+    if (repositories.length > 1) {
+      throw new Error(
+        `There are more than 1 active repositories. Please close unwanted deployments.`
+      );
+    }
+
+    return repositories[0];
+  }
+
+  private async closeRepository(repositoryId: string): Promise<void> {
+    const response = await fetch(`${NEXUS_API_BASE_URL}/bulk/close`, {
+      headers: this.getNexusRequestHeaders(),
+      method: 'POST',
+      body: JSON.stringify({
+        data: { stagedRepositoryIds: [repositoryId] },
+      }),
+    });
+
+    if (!response.ok) {
+      throw new Error(
+        `Unable to close repository: ${response.status}, ${response.statusText}`
+      );
+    }
+
+    const poolingStartTime = Date.now();
+
+    while (true) {
```

Comment:
Can we benefit from [`withRetry`](https://github.com/getsentry/craft/blob/2a0eda0b9921dca762090b3fa357e03b56c2f66b/src/utils/async.ts#L155) in this whole block?

Reply:
We could if those servers would be slightly more reliable. `withRetry` is based on a number of attempts, whereas here, we are using time-based deadline. With attempts-based and this servers, it could take between 10 and 100min for the same number of attempts 🥲